fsspec/implementations/cached.py

```diff
@@ -834,3 +833,19 @@ def hash_name(path, same_name):
     else:
         hash = hashlib.sha256(path.encode()).hexdigest()
     return hash
+
+
+@contextlib.contextmanager
+def atomic_write(path, mode="wb"):
```

Comment:
Looks great to me, thank you!  I  would have only appreciated a docstring briefly describing the purpose/implementation of this function.